src/main/scala/nl/knaw/dans/easy/ingest/EasyIngest.scala

```diff
@@ -179,32 +180,31 @@ object EasyIngest {
 
   private def pidToUri(pid: String): String = s"info:fedora/$pid"
 
-  private def addDataStream(sdo: File, dsSpec: DatastreamSpec, pidDictionary: PidDictionary): Try[URI] = Try {
+  private def addDataStream(sdo: File, dsSpec: DatastreamSpec, pid: Pid): Try[URI] = Try {
     log.debug(s"Getting datastreamId from spec: $dsSpec")
-    val datastreamId = (dsSpec.dsID, dsSpec.contentFile, dsSpec.dsLocation) match {
-      case (id, _, _) if id != "" => id
-      case ("", file, _) if file != "" => file
+    val datastreamId = (dsSpec.dsID, dsSpec.contentFile) match {
+      case (id, _) if id.nonEmpty => id
+      case ("", file) if file.nonEmpty => file
       case _ => throw new RuntimeException(s"Invalid datastream specification provided in ${sdo.getName}")
     }
 
     log.debug(s"Adding datastream with dsId = $datastreamId")
-    var request = addDatastream(pidDictionary(sdo.getName), datastreamId).mimeType(dsSpec.mimeType).controlGroup(dsSpec.controlGroup)
+    var request = addDatastream(pid, datastreamId).mimeType(dsSpec.mimeType).controlGroup(dsSpec.controlGroup)
```

Comment:
I did not like the `var` either, but now reassignments are at least reduced to one. Considered using two `val`'s but that might lead to confusion in case of future fixes/enhancements and using the wrong version.